pkg/authn/signingMethod.go

```diff
@@ -0,0 +1,91 @@
+package authn
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"errors"
+	"math/big"
+
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const (
+	ALGORITHM  = "ES256K"
+	SIG_LENGTH = 65
+	R_LENGTH   = 32
+	S_LENGTH   = 32
+)
+
+var (
+	ErrWrongKeyFormat = errors.New("wrong key type")
+	ErrBadSignature   = errors.New("bad signature")
+	ErrVerification   = errors.New("signature verification failed")
+)
+
+/*
+*
+The JWT signing method for secp256k1. Inspired by https://github.com/ureeves/jwt-go-secp256k1/blob/master/secp256k1.go
+but updated to work with the latest version of jwt-go.
+*/
+type SigningMethodSecp256k1 struct{}
+
+func (sm *SigningMethodSecp256k1) Verify(signingString string, sig []byte, key interface{}) error {
+	pub, ok := key.(*ecdsa.PublicKey)
+	if !ok {
+		return ErrWrongKeyFormat
+	}
+
+	hashedString := hashString(signingString)
+
+	if len(sig) != SIG_LENGTH {
+		return ErrBadSignature
+	}
+
+	r := new(big.Int).SetBytes(sig[:R_LENGTH])                    // R
+	s := new(big.Int).SetBytes(sig[R_LENGTH : R_LENGTH+S_LENGTH]) // S
+
+	if !ecdsa.Verify(pub, hashedString, r, s) {
+		return ErrVerification
+	}
+
+	return nil
+}
+
+func (sm *SigningMethodSecp256k1) Sign(signingString string, key interface{}) ([]byte, error) {
+	priv, ok := key.(*ecdsa.PrivateKey)
+	if !ok {
+		return nil, ErrWrongKeyFormat
+	}
+
+	hashedString := hashString(signingString)
+
+	sig, err := ethcrypto.Sign(hashedString, priv)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sig) != SIG_LENGTH {
+		return nil, ErrBadSignature
+	}
+
+	return sig, nil
+}
+
+func (sm *SigningMethodSecp256k1) Alg() string {
+	return ALGORITHM
+}
+
+func hashString(signingString string) []byte {
+	hasher := crypto.SHA256.New()
```

Comment:
We use `keccak256` in `registrant.go`, which I think is also consistent with ERC191 and personal_sign. Would it make sense to share the same ciphersuite?

Reply:
If we do this, one thing we need to be careful about is having good domain separation, so that signatures from one use-case cannot be applied to the other.

Reply:
That’s a good call. I can change out the hash function.

Reply:
> If we do this, one thing we need to be careful about is having good domain separation, so that signatures from one use-case cannot be applied to the other.

It should be impossible to have a valid payload that is both a JWT and an UnsignedOriginatorEnvelope, but we could add some prefix to both payloads before hashing so we never have to think about it again.

Reply:
I added a domain separation label just to make sure it's truly impossible to re-use a signature (outside of hash collisions)